### ISSUE TITLE
[Bug] Extension does not work when project is open from context menu on Manjaro Linux

### ISSUE THREAD

**Issue by JBudny** [microsoft/vscode-react-native#2084]
# 🐛 What is the bug? How can we reproduce it?

I am uncertain about the origin of this bug, whether it is from the extension, Visual Studio Code, or my system. However, I will share it here in case others encounter similar issues. I discovered that the reason for my problems was that I opened my project folder in Visual Studio Code using the context menu option, as shown in the image below.
![image](https://github.com/microsoft/vscode-react-native/assets/32987100/e9aeaccd-29a2-43ac-ab01-121ca50e5d44)
Once I identified the source of the issues, I attempted to open my project by entering `code .` in both the `Zsh` and `Bash` shell and found out that it functions correctly.

The issue:
I was trying to debug my React Native app using the `Run and Debug` option in Visual Studio Code. I made sure to properly set up my React Native environment, Android Studio, PATH, and `launch.json`. Additionally, I cleared the Metro cache by running `react-native start --reset-cache`, cleaned the Android project cache using the `Clean Project` option in Android Studio and `gradlew clean --refresh-dependencies`, and deleted the Gradle cache from my $HOME. Despite these efforts, I was unable to successfully initiate the debugger and to use some other options listed below.

## Expected behavior
I was expecting these options to run properly
1. `React Native: Launch Android Emulator`
2. `React Native: Check development environment configuration`
3. `Run and Debug`

## Debug output
I was getting these outputs, respectively:

1. `React Native: Launch Android Emulator`

<details>
<summary>Debug Console</summary>

```text
PASTE OUTPUT OF DEBUG CONSOLE (View -> Toggle Debug Console)
```
</details>

<details>
<summary>React Native output channel</summary>

```text
[Error] An error occurred while starting the Android emulator (error code 305)

Stack: Error: An error occurred while starting the Android emulator (error code 305): Error while executing command 'emulator -list-avds': Command failed: emulator -list-avds
/bin/sh: line 1: emulator: command not found

	at o.getWrappedError (/home/xxx/.vscode/extensions/msjsdiag.vscode-react-native-1.12.2/dist/rn-extension.js:21:382468)
	at i.wrapError (/home/xxx/.vscode/extensions/msjsdiag.vscode-react-native-1.12.2/dist/rn-extension.js:21:374820)
	at /home/xxx/.vscode/extensions/msjsdiag.vscode-react-native-1.12.2/dist/rn-extension.js:21:374389
```
</details>

<details>
<summary>Developer Tools console</summary>

```text
PASTE OUTPUT OF DEVELOPER TOOLS CONSOLE (Help -> Toggle Developer Tools -> Select Console tab)
```
</details>

2. `React Native: Check development environment configuration`

<details>
<summary>Debug Console</summary>

```text
PASTE OUTPUT OF DEBUG CONSOLE (View -> Toggle Debug Console)
```
</details>

<details>
<summary>React Native output channel</summary>

```text
[Info] Starting Environment check...

<<< Dev Environment verification result >>>

*** Common ***
 ✓ Node.JS
 ✓ NPM

*** Android ***
 ✖ ADB - Required for app installation. Minimal version is 12
  Command not found. Make sure ADB is installed
 ✖ Android Emulator - Required for working with Android emulators
  Command not found. Make sure Android Emulator is installed
 ✖ Android Env - Required for launching React Native apps
  "ANDROID_SDK_ROOT" not found in path. Ensure all variables are set up accroding to this guide - https://reactnative.dev/docs/environment-setup
 ✓ Java
 ✖ Gradle - Requried for building android apps
  Command not found. Make sure Gradle is installed
```
</details>

<details>
<summary>Developer Tools console</summary>

```text
  ERR TextModelPart is disposed!: Error: TextModelPart is disposed!
    at d.g (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:121:22451)
    at d.getWordAtPosition (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:631:119008)
    at W.getWordAtPosition (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:736:9056)
    at k.h (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:68470)
    at new $ (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:68345)
    at new k (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:69207)
    at D (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:69734)
    at O.G (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:74294)
    at O.F (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:74091)
    at new O (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:70761)
    at q (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:75166)
    at u.value (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:75311)
    at f.y (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:87:1902)
    at f.z (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:87:1972)
    at f.fire (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:87:2188)
    at x.setModel (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:744:400)
    at k.setInput (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:2396:11447)
    at async k.setInput (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:2400:3886)
log.ts:441   ERR TextModelPart is disposed!: Error: TextModelPart is disposed!
    at d.g (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:121:22451)
    at d.getWordAtPosition (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:631:119008)
    at W.getWordAtPosition (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:736:9056)
    at k.h (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:68470)
    at new $ (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:68345)
    at new k (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:69207)
    at D (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:69734)
    at O.G (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:74294)
    at O.F (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:74091)
    at u.value (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:760:70483)
    at f.y (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:87:1902)
    at f.z (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:87:1972)
    at f.fire (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:87:2188)
    at x.setModel (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:744:400)
    at k.setInput (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:2396:11447)
    at async k.setInput (vscode-file://vscode-app/opt/visual-studio-code/resources/app/out/vs/workbench/workbench.desktop.main.js:2400:3886)
console.ts:137 [Extension Host] TypeError: Cannot read properties of undefined (reading 'uri')
	at /home/xxx/.vscode/extensions/dotenv.dotenv-vscode-0.28.1/lib/autocloaking.js:30:55
	at Array.filter (<anonymous>)
	at h.value (/home/xxx/.vscode/extensions/dotenv.dotenv-vscode-0.28.1/lib/autocloaking.js:28:57)
	at c.y (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
	at c.z (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:80:1972)
	at c.fire (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:80:2188)
	at h.value (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:127:8509)
	at c.y (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
	at c.fire (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
	at l.acceptDocumentsAndEditorsDelta (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:127:12473)
	at l.$acceptDocumentsAndEditorsDelta (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:127:10970)
	at n.S (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:134:10836)
	at n.Q (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:134:10602)
	at n.M (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:134:9686)
	at n.L (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:134:8762)
	at h.value (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:134:7565)
	at c.y (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
	at c.fire (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
	at d.fire (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:96:14047)
	at h.value (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:153:7918)
	at c.y (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
	at c.fire (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
	at d.fire (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:96:14047)
	at MessagePortMain.<anonymous> (/opt/visual-studio-code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:153:6198)
	at MessagePortMain.emit (node:events:513:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:367)
```
</details>

3. `Run and Debug`

<details>
<summary>Debug Console</summary>

```text
Launching the application
An error occurred while launching the application. Error while executing command '/home/xxx/git/XxX/node_modules/.bin/react-native run-android --no-packager' (error code 101) (error code 303)
```
</details>

<details>
<summary>React Native output channel</summary>

```text
[Warning] Error: Packager is still running. If the packager was started outside VS Code, please quit the packager process using the task manager.

[Info] Packager is already running.

[Warning] Error: React Native Packager running outside of VS Code. If you want to debug please use the 'Attach to packager' option.

[Info] Prewarming bundle cache. This may take a while ...

[Info] About to get: http://localhost:8081/index.bundle?platform=android

[Warning] The Bundle Cache was prewarmed.

[Info] Building and running application.

[Error] Error: Error while executing command '/home/xxx/git/XxX/node_modules/.bin/react-native run-android --no-packager' (error code 101)

[Error] Error: Error while executing command '/home/xxx/git/XxX/node_modules/.bin/react-native run-android --no-packager' (error code 101)
```
</details>

<details>
<summary>Developer Tools console</summary>

```text
ERR [Extension Host] An error occurred while launching the application. Error while executing command '/home/xxx/git/XxX/node_modules/.bin/react-native run-android --no-packager' (error code 101) (error code 303)
console.ts:137 [Extension Host] An error occurred while launching the application. Error while executing command '/home/xxx/git/XxX/node_modules/.bin/react-native run-android --no-packager' (error code 101) (error code 303)
w @ console.ts:137
$logExtensionHostMessage @ mainThreadConsole.ts:39
S @ rpcProtocol.ts:456
Q @ rpcProtocol.ts:441
M @ rpcProtocol.ts:371
L @ rpcProtocol.ts:297
(anonymous) @ rpcProtocol.ts:161
y @ event.ts:1140
fire @ event.ts:1171
fire @ ipc.net.ts:650
Q.onmessage @ localProcessExtensionHost.ts:581

```
</details>

## Environment

Please tell us about your system and your project:

* `npx react-native doctor` or `npx expo doctor` if your project is using Expo:

```text
DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Common
 ✓ Node.js - Required to execute JavaScript code
 ✓ yarn - Required to install NPM dependencies
 ✓ Metro - Required for bundling the JavaScript code

Android
 ✓ Adb - Required to verify if the android device is attached correctly
 ✓ JDK - Required to compile Java code
 ✓ Android Studio - Required for building and installing your app on Android
 ✓ ANDROID_HOME - Environment variable that points to your Android SDK installation
 ✓ Android SDK - Required for building and installing your app on Android

Errors:   0
Warnings: 0
```

* `envinfo`:

```text
System:
    OS: Linux 6.6 Manjaro Linux
    CPU: (20) x64 13th Gen Intel(R) Core(TM) i5-13600KF
    Memory: 11.37 GB / 31.19 GB
    Container: Yes
    Shell: 5.9 - /usr/bin/zsh
  Binaries:
    Node: 21.4.0 - ~/.nvm/versions/node/v21.4.0/bin/node
    Yarn: 1.22.21 - ~/.nvm/versions/node/v21.4.0/bin/yarn
    npm: 10.2.4 - ~/.nvm/versions/node/v21.4.0/bin/npm
    pnpm: 8.12.1 - ~/.nvm/versions/node/v21.4.0/bin/pnpm
    Watchman: 20231013.222836.0 - /usr/bin/watchman
  Managers:
    pip3: 23.3.1 - /usr/bin/pip3
  Utilities:
    CMake: 3.28.0 - /usr/bin/cmake
    Make: 4.4.1 - /usr/bin/make
    GCC: 13.2.1 - /usr/bin/gcc
    Git: 2.43.0 - /usr/bin/git
    Clang: 16.0.6 - /usr/bin/clang
    Ninja: 1.11.1 - /usr/bin/ninja
    FFmpeg: 6.1 - /usr/bin/ffmpeg
    Curl: 8.5.0 - /usr/bin/curl
  Virtualization:
    Docker: 24.0.7 - /usr/bin/docker
    VirtualBox: 7.0.12 - /usr/bin/vboxmanage
  SDKs:
    Android SDK:
      API Levels: 30, 33, 34
      Build Tools: 30.0.2, 30.0.3, 33.0.0, 34.0.0
      System Images: android-33 | Google APIs Intel x86_64 Atom, android-33 | Google Play Intel x86_64 Atom, android-34 | Google Play Intel x86_64 Atom
  IDEs:
    Android Studio: AI-223.8836.35.2231.11005911
    Nano: 7.2 - /usr/bin/nano
    VSCode: 1.84.2 - /usr/bin/code
    Vim: 9.0 - /usr/bin/vim
  Languages:
    Bash: 5.2.21 - /usr/bin/bash
    Java: 17.0.9 - /usr/bin/javac
    Perl: 5.38.1 - /usr/bin/perl
    Protoc: 25.1 - /usr/bin/protoc
    Python: 3.11.6 - /usr/bin/python
    Python3: 3.11.6 - /usr/bin/python3
  Databases:
    SQLite: 3.44.2 - /usr/bin/sqlite3
  Browsers:
    Brave Browser: 120.1.61.104
```

* React Native Tools extension version: 1.12.2
* Expo SDK version (if applicable): VERSION

**Comment by EzioLi01** [comment]
Hey @JBudny , thanks for reaching us. 
Could you please try to use RN native command `react-native run-android` to see if it can open your emulator successfully?

**Comment by JBudny** [comment]
> Hey @JBudny , thanks for reaching us. Could you please try to use RN native command `npx react-native run-android` to see if it can open your emulator successfully?

Hello, both `npx react-native run-android` and `node_modules/.bin/react-native run-android --no-packager` work fine for me. They successfully open the emulator and run the app without any problems, whether I open the project directory in Visual Studio Code through the context menu or the terminal. I also utilize the `DiemasMichiels.emulate` extension to open emulators and it works fine as well.

**Comment by EzioLi01** [comment]
OK, got it. Thanks for the info. Let me summarize the scenario. You are using `Command Palette` to run emulator in Linux as below. Emulator cannot be found. If I'm misunderstanding anything, please correct me.
I will look into this and any update will comment here.

![image](https://github.com/microsoft/vscode-react-native/assets/19662219/aa687a0e-bb2d-4c4d-b6cb-d1517ec7ac7d)

**Comment by JBudny** [comment]
Thank you for summarizing, you are correct for the most part. I just wanted to mention one small correction.

> You are using Command Palette

that's right
> BTW, only open project from context menu has this issue?

exactly
> Emulator cannot be found

You are almost there.  It's not the emulator that cannot be found, but the emulator command as this error from the output channel says:
```text
Stack: Error: An error occurred while starting the Android emulator (error code 305): Error while executing command 'emulator -list-avds': Command failed: emulator -list-avds
/bin/sh: line 1: emulator: command not found
```

> Could you please try to run emulator -list-avds manually in your local terminal to see if it works?

Sure. It recognizes the command and returns a list of emulators properly on both `Zsh` and `Bash` shells. I have tested it in both the integrated terminal in Visual Studio Code opened from the context menu, as well as in a separate terminal.

Thank you for addressing this issue. Feel free to ask if you have any other questions

**Comment by EzioLi01** [comment]
Start investigating..

**Comment by EzioLi01** [comment]
I can repo this issue on my side, but I think it's should not be an issue in extension. 

I have done some tests:
1. Add sample js file and add npm script in package.json to execute it
2. Create launch settings to run file via npm

Result: 
1. Only use `code` command in terminal to open vscode can execute script well
2. If using context menu or application launcher open vscode, npm command cannot be found

I will try to get workaround for this and also report this to vscode team for investigation.

**Comment by JBudny** [comment]
Hey. Thanks for spending your time on this issue, I appreciate it! I can live with the workaround using the `code` command, it's not a major issue for me. But I wanted to mention this, just in case someone else is going through the same thing and because I just happened to find this correlation by chance. I am closing this issue as I saw you already reported it on the vscode repository. 

Once again, thank you for your help!
